### PR TITLE
Test/backend test cases for fAIr

### PR DESCRIPTION
**Pull request by nifedara**
### What does this PR do?
To solve the issue: [Implement Backend Test Cases for fAIr](https://github.com/hotosm/fAIr/issues/229)

### Consideration
- Adds factories to isolate the test data
- Excludes `callback` endpoint from coverage
- Increases coverage from 64% to 76%

image

### CODE REVIEW

**Comment by nifedara** [comment]
Happy Monday, @kshitijrajsharma. Kindly review this. 
Thanks

**Comment by kshitijrajsharma** [comment]
Do you want to add github action .yaml  on this PR ? or sep one ?

File: backend/tests/test_endpoints.py [hunk 1]
```diff
@@ -43,104 +59,254 @@ def test_create_dataset(self):
...
+        }
         res = self.client.post(
-            f"{API_BASE}/dataset/image/build/",
-            json.dumps(build_dt_payload),
+            f"{API_BASE}/training/",
+            json.dumps(training_payload),
             headers=json_type_header,
         )
-        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        print(res.json())
```
Comment (kshitijrajsharma) [inline backend/tests/test_endpoints.py]: no need to print

File: backend/tests/test_endpoints.py [hunk 2]
```diff
@@ -43,104 +59,254 @@ def test_create_dataset(self):
...
+            "epochs": 20,
+            "batch_size": 8,
+            "zoom_level": [19, 20],
+        }
+        res = self.client.post(
+            f"{API_BASE}/feedback/training/submit/",
+            json.dumps(training_feedback_payload),
+            headers=self.json_type_header,
+        )
+        # submit unfished/unpublished training feedback should not pass
```
Comment (kshitijrajsharma) [inline backend/tests/test_endpoints.py]: unfinished typo ---

File: backend/tests/test_endpoints.py [hunk 3]
```diff
@@ -43,104 +59,254 @@ def test_create_dataset(self):
...
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        # generate feedback aoi GPX view - feedback aoi_id
+
+        res = self.client.get(
+            f"{API_BASE}/feedback-aoi/gpx/{feedbackAoi.id}/", headers=headersList
+        )
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+    def test_get_workspace(self):
```
Comment (kshitijrajsharma) [inline backend/tests/test_endpoints.py:306]: add one more for /workspace/download , test-download-workspace

**Review by kshitijrajsharma — APPROVED** [review]

**Comment by kshitijrajsharma** [comment]
Kindly rebase your branch and now afterwards  test cases should be running

**Pull request merged**